src/js/controllers/LoginController.js

```diff
@@ -19,17 +19,44 @@ function LoginController($scope, $http, $state, AuthService) {
         AuthService.login($scope.account.email, $scope.account.password, $scope.account.remember,
           (response) => {
               if (response === 'SUCCESS') {
-                  $state.go('home');
+
+                  // If we're logging in through the modal, don't redirect.
+                  if (!$rootScope.modals.loginModalOpen) {
+                      $state.go('home');
+                  } else {
+                      $rootScope.modals.loginModalClose();
+                  }
+
               } else if (response === 'FAILURE') {
                   $scope.loginForm.general.incorrect = true;
               } else if (response === 'ERROR') {
                   $scope.loginForm.general.issue = true;
               }
+
+              $scope.loginSubmitting = false;
           }
         );
+    };
 
-        $scope.loginSubmitting = false;
+    // Login modal
+    $rootScope.modals.loginModal = ($event) => {
+        if (!$rootScope.modals.loginModalOpen) {
+            $rootScope.modals.loginModalOpen = true;
+            $rootScope.modals.loginModal = $mdDialog.show({
```

Comment:
how exactly does this work? Youre assigning a function to `$rootScope.modals.loginModal`, but then in that function you reassign it. Does `$mdDialog.show()` return a `function($event)`? Even still, this sort of pattern seems, to me, like it could lead to problems. It is hard to figure out what `loginModal` is at any given time, to me.